script.js

```diff
@@ -1,4 +1,632 @@
+  {
+    name: "8",
+    suit: "diamonds",
+    score: 8,
+    image: "assets/8D.png",
+  },
+  {
+    name: "9",
+    suit: "diamonds",
+    score: 9,
+    image: "assets/9D.png",
+  },
+  {
+    name: "10",
+    suit: "diamonds",
+    score: 10,
+    image: "assets/10D.png",
+  },
+  {
+    name: "jack",
+    suit: "diamonds",
+    score: 10,
+    image: "assets/JD.png",
+  },
+  {
+    name: "queen",
+    suit: "diamonds",
+    score: 10,
+    image: "assets/QD.png",
+  },
+  {
+    name: "king",
+    suit: "diamonds",
+    score: 10,
+    image: "assets/KD.png",
+  },
+  {
+    name: "ace",
+    suit: "clubs",
+    score: 1,
+    image: "assets/AC.png",
+  },
+  {
+    name: "2",
+    suit: "clubs",
+    score: 2,
+    image: "assets/2C.png",
+  },
+  {
+    name: "3",
+    suit: "clubs",
+    score: 3,
+    image: "assets/3C.png",
+  },
+  {
+    name: "4",
+    suit: "clubs",
+    score: 4,
+    image: "assets/4C.png",
+  },
+  {
+    name: "5",
+    suit: "clubs",
+    score: 5,
+    image: "assets/5C.png",
+  },
+  {
+    name: "6",
+    suit: "clubs",
+    score: 6,
+    image: "assets/6C.png",
+  },
+  {
+    name: "7",
+    suit: "clubs",
+    score: 7,
+    image: "assets/7C.png",
+  },
+  {
+    name: "8",
+    suit: "clubs",
+    score: 8,
+    image: "assets/8C.png",
+  },
+  {
+    name: "9",
+    suit: "clubs",
+    score: 9,
+    image: "assets/9C.png",
+  },
+  {
+    name: "10",
+    suit: "clubs",
+    score: 10,
+    image: "assets/10C.png",
+  },
+  {
+    name: "jack",
+    suit: "clubs",
+    score: 10,
+    image: "assets/JC.png",
+  },
+  {
+    name: "queen",
+    suit: "clubs",
+    score: 10,
+    image: "assets/QC.png",
+  },
+  {
+    name: "king",
+    suit: "clubs",
+    score: 10,
+    image: "assets/KC.png",
+  },
+  {
+    name: "ace",
+    suit: "spades",
+    score: 1,
+    image: "assets/AS.png",
+  },
+  {
+    name: "2",
+    suit: "spades",
+    score: 2,
+    image: "assets/2S.png",
+  },
+  {
+    name: "3",
+    suit: "spades",
+    score: 3,
+    image: "assets/3S.png",
+  },
+  {
+    name: "4",
+    suit: "spades",
+    score: 4,
+    image: "assets/4S.png",
+  },
+  {
+    name: "5",
+    suit: "spades",
+    score: 5,
+    image: "assets/5S.png",
+  },
+  {
+    name: "6",
+    suit: "spades",
+    score: 6,
+    image: "assets/6S.png",
+  },
+  {
+    name: "7",
+    suit: "spades",
+    score: 7,
+    image: "assets/7S.png",
+  },
+  {
+    name: "8",
+    suit: "spades",
+    score: 8,
+    image: "assets/8S.png",
+  },
+  {
+    name: "9",
+    suit: "spades",
+    score: 9,
+    image: "assets/9S.png",
+  },
+  {
+    name: "10",
+    suit: "spades",
+    score: 10,
+    image: "assets/10S.png",
+  },
+  {
+    name: "jack",
+    suit: "spades",
+    score: 10,
+    image: "assets/JS.png",
+  },
+  {
+    name: "queen",
+    suit: "spades",
+    score: 10,
+    image: "assets/QS.png",
+  },
+  {
+    name: "king",
+    suit: "spades",
+    score: 10,
+    image: "assets/KS.png",
+  },
+];
```

Comment:
Understood from our last conversation that it can be tricky to use a deck generator function to set the filenames of the assets. As mentioned the last time, you can still use a deck generator function only if you are willing to change the names of the files.

In fact, perhaps you can still keep the existing filenames and still set the image path strings using a generator. Do you realise there's a pattern to how the image path names are assigned? It takes the first letter (capitalised) of the value in the `name` key if it's not a number, otherwise it takes the name as is (if it's a number). Then it is appended with the first letter capitalised of the suit name.

For the King of Spades, notice this:
```
{
  name: "king",
  suit: "spades",
  score: 10,
  image: "assets/K(ing)S(pades).png",
}
```

and 10 of Hearts:
```
{
  name: "10",
  suit: "hearts",
  score: 10,
  image: "assets/10H(earts).png",
}
```

Thus, if you are generating the deck using a deck generator function (and loops), you can do so programmatically with `if`s or `else`s:

```
// aces, kings, jacks, queens
// their card names are not numbers
if (Number.isNaN(Number(card.name))) {
  // appends "assets/" with the first letter of 'name' capitalised
  // then appends first letter of 'suit' capitalised
  card.image = "assets/" + card.name.charAt(0).toUpperCase() + card.suit.charAt(0).toUpperCase()
} else {
  // Number cards, can just append the entire name of card
  // Then add first letter of suit capitalised.
  card.image = "assets/" + card.name + card.suit.charAt(0).toUpperCase()
}
```